configurator/src/main/java/com/devonfw/tools/ide/migrator/Migrations.java

```diff
@@ -111,7 +111,25 @@ public static MigrationImpl devon4j() {
             "implements $1Dao, io.oasp.module.jpa.common.base.LegacyDaoQuerySupport<$1Entity>",
             FileFilterPattern.reject("Application(MasterData)?DaoImpl\\.java")) //
         .and() //
-        .next().build();
+        .next().to(VersionIdentifier.ofDevon4j("3.1.0")) //
+        .pom().replaceProperty("devon4j.version", "3.1.0") //
+        .replaceProperty("spring.boot.version", "2.1.4.RELEASE") //
+        // only for oasp4j legacy project (flyway.version not present in projects created from devon4j)
+        .replaceProperty("flyway.version", "5.2.4") //
+        .and().java()
+        .replace("import org.dozer.DozerBeanMapper;", "import com.github.dozermapper.core.DozerBeanMapper;")
+        .replace("import org.dozer.Mapper;",
+            "import com.github.dozermapper.core.DozerBeanMapperBuilder;\n import com.github.dozermapper.core.Mapper;")
+        .replace("return new DozerBeanMapper(beanMappings);",
+            "Mapper mapper = DozerBeanMapperBuilder.create().withMappingFiles(beanMappings).build();\n return mapper;")//
+        .and().xml("dozer-mapping.xml")
+        .replaceNamespace("http://dozer.sourceforge.net", "http://dozermapper.github.io/schema/bean-mapping")
+        .replaceNamespace("http://dozer.sourceforge.net           http://dozer.sourceforge.net/schema/beanmapping.xsd",
```

Comment:
this looks borken or not reliable. There are two URLs concated with a specific number of whitespaces.

Reply:
Seems to relate to schemalocation
https://github.com/devonfw/devon4j/blob/97ac9538ed4f910d9e17ee30def646d5cce7d825/templates/server/src/main/resources/archetype-resources/core/src/main/resources/config/app/common/dozer-mapping.xml#L5

This may also be reformatted or contain newlines. Hence IMHO we need to define a replacement rule based on a regex matching any number of whitespaces including newlines (`\s+`).